exporters/otlp/trace/src/main/java/io/opentelemetry/exporter/otlp/trace/OtlpHttpSpanExporter.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.trace;
+
+import com.google.rpc.Status;
+import io.opentelemetry.api.metrics.BoundLongCounter;
+import io.opentelemetry.api.metrics.GlobalMeterProvider;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.exporter.otlp.internal.SpanAdapter;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.internal.ThrottlingLogger;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.concurrent.ThreadSafe;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import org.jetbrains.annotations.NotNull;
+
+/** Exports spans using OTLP via HTTP, using OpenTelemetry's protobuf model. */
+@ThreadSafe
+public final class OtlpHttpSpanExporter implements SpanExporter {
+
+  private static final String EXPORTER_NAME = OtlpHttpSpanExporter.class.getSimpleName();
+  private static final Labels EXPORTER_NAME_LABELS = Labels.of("exporter", EXPORTER_NAME);
+  private static final Labels EXPORT_SUCCESS_LABELS =
+      Labels.of("exporter", EXPORTER_NAME, "success", "true");
+  private static final Labels EXPORT_FAILURE_LABELS =
+      Labels.of("exporter", EXPORTER_NAME, "success", "false");
+
+  private final ThrottlingLogger logger =
+      new ThrottlingLogger(Logger.getLogger(OtlpHttpSpanExporter.class.getName()));
+
+  private final BoundLongCounter spansSeen;
+  private final BoundLongCounter spansExportedSuccess;
+  private final BoundLongCounter spansExportedFailure;
+
+  private final OkHttpClient client;
+  private final String endpoint;
+  private final Map<String, List<String>> headers;
+  private final RequestResponseHandler requestResponseHandler;
+
+  OtlpHttpSpanExporter(
+      OkHttpClient client,
+      String endpoint,
+      Map<String, List<String>> headers,
+      RequestResponseHandler requestResponseHandler) {
+    // TODO: should this be io.opentelemetry.exporters.otlp with a label that indicates the
+    // protocol?
```

Comment:
Thoughts?

Reply:
well, this is just the meter name, so having include the protocol seems correct, I think. I wouldn't imagine someone would end up having both exporters in the same JVM so breaking out by label doesn't seem to add much value.